content/blog/2024/11/04/2024-11-04-jenkins-tutorial-revamp.adoc

```diff
@@ -0,0 +1,273 @@
+We'll examine how we transformed complex setup procedures into streamlined, accessible guides using modern tooling and best practices.
+The discussion will cover the challenges we encountered, the technical solutions we implemented, and the measurable improvements achieved through these changes.
+
+== The Problem: Old Jenkins Tutorials
+
+=== A Visual Comparison
+
+Before we dive into the specifics, let's take a visual journey through the old and new tutorial experiences:
+
+video::jVOpNnOXr_s[youtube,width=839]
+
+Did you notice how much scrolling was required on the right side?
+It was like a never-ending story.
+And did you see the amount of code users had to copy-paste?
+While we'll always need a certain quantity of code for the Jenkins pipeline itself, the setup process should be as simple as possible.
+That wasn't the case with the old tutorials.
+
+=== Breaking Down the Issues
+
+Our old tutorials, while comprehensive, presented several significant challenges:
+
+1. *Overwhelming complexity*: Users were bombarded with Docker commands before they could even start learning about Jenkins.
+The tutorial assumed users were Docker gurus, throwing commands at them like there's no tomorrow.
+
+2. *Assumptions about user knowledge*: The tutorials assumed a high level of Docker expertise.
+But let me ask you this: how many of you knew Docker really well the first time you tried Jenkins?
+That's not a fair expectation.
+
+3. *Lengthy setup process*: A significant portion of the tutorial was dedicated to setup rather than learning Jenkins itself.
+We were spending more time with Docker than with Jenkins!
+
+4. *Reliance on outdated tools*: We were heavily dependent on the deprecated BlueOcean plugin, which was no longer recommended.
+
+5. *Docker in Docker*: The old tutorial used Docker in Docker, which is considered a bad practice from a security standpoint.
+
+6. *Lack of context*: The tutorials often lacked explanations for why certain steps were necessary, leaving users to blindly follow instructions without understanding.
+
+Here's a snippet from the old Maven tutorial to illustrate these points:
+
+[source,bash]
+----
+docker network create jenkins
+docker run \
+  --name jenkins-docker \
+  --rm \
+  --detach \
+  --privileged \
+  --network jenkins \
+  --network-alias docker \
+  --env DOCKER_TLS_CERTDIR=/certs \
+  --volume jenkins-docker-certs:/certs/client \
+  --volume jenkins-data:/var/jenkins_home \
+  --publish 2376:2376 \
+  docker:dind \
+  --storage-driver overlay2
+----
+
+This command, while functional, is overwhelming for beginners and lacks context.
+Why do we need a Docker network?
+Why are we running Docker in Docker?
+These questions were left unanswered.
+
+== The Solution: Revamping with Docker Compose
+
+After much deliberation and feedback, we decided to leverage Docker Compose to simplify the entire setup process. 
```

Comment:
```suggestion
After much deliberation and feedback, we decided to leverage link:https://docs.docker.com/compose/intro/compose-application-model/[Docker Compose] to simplify the entire setup process. 
```
The link could be added here to provide resources to the reader. Not necessary to implement though.